test/e2e/framework/dataplane.go

```diff
@@ -125,3 +141,74 @@ func execCmdInBash(p tcp.ConnectivityTestParams, cmd []string, pod *v1.Pod) (str
 
 	return p.Framework.ExecWithOptions(execOptions, p.FromCluster)
 }
+
+func createTCPHeadlessService(cluster framework.ClusterIndex, selectorName string, port int, namespace string) *v1.Service {
+	tcpService := v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("test-headless-svc-%s", selectorName),
+		},
+		Spec: v1.ServiceSpec{
+			Type:      "ClusterIP",
+			ClusterIP: v1.ClusterIPNone,
+			Ports: []v1.ServicePort{{
+				Port:       int32(port),
+				Name:       "tcp",
+				TargetPort: intstr.FromInt(port),
+				Protocol:   v1.ProtocolTCP,
+			}},
+			Selector: map[string]string{
+				testAppLabel: selectorName,
+			},
+		},
+	}
+
+	services := framework.KubeClients[cluster].CoreV1().Services(namespace)
+
+	return framework.AwaitUntil("create service", func() (interface{}, error) {
+		service, err := services.Create(context.TODO(), &tcpService, metav1.CreateOptions{})
+		if errors.IsAlreadyExists(err) {
+			err = services.Delete(context.TODO(), tcpService.Name, metav1.DeleteOptions{})
+			if err != nil {
+				return nil, err
+			}
+
+			service, err = services.Create(context.TODO(), &tcpService, metav1.CreateOptions{})
+		}
+
+		return service, err
+	}, framework.NoopCheckResult).(*v1.Service)
```

Comment:
You should be able to use the `util.CreateAnew` function in admiral.

Reply:
> You should be able to use the `util.CreateAnew` function in admiral.

`CreateAnew` takes `resource.Interface` as client argument while services here is `ServiceInterface` which is just a generic interface. Also, `CreateAnew` doesn't return the created object.

Reply:
> `CreateAnew` takes `resource.Interface` as client argument while services here is `ServiceInterface` which is just a generic interface. 

You can use `resource.ForService` to get a Service `resource.Interface`.

> Also, `CreateAnew` doesn't return the created object [which not an issue for now as we don't use any generated fields]

https://github.com/submariner-io/admiral/pull/253

Reply:
Actually if we put this in shipyard then we can't use admiral. https://github.com/submariner-io/shipyard/pull/595 adds a function for this so we can use that.